Doc/library/venv.rst

```diff
@@ -170,11 +170,11 @@ creation according to their needs, the :class:`EnvBuilder` class.
 
     .. method:: ensure_directories(env_dir)
 
-        Creates the environment directory and all necessary directories, and
-        returns a context object.  This is just a holder for attributes (such as
-        paths), for use by the other methods. The directories are allowed to
-        exist already, as long as either ``clear`` or ``upgrade`` were
-        specified to allow operating on an existing environment directory.
+        Creates the environment directory and all necessary subdirectories that
+        don't already exist, and returns a context object.  This is just a
+        holder for attributes (such as paths), for use by the other methods.
+        Any existing environment directories will remain unaffected as long as
```

Comment:
I suggest changing the sentence to:

If the :class:`EnvBuilder` is created with the arg ``clear=True``, contents of the environment directory will be cleared and then all necessary subdirectories will be recreated.

-- this describes the effect of the method more fully and given the description, there is no reason for the reader to think that existing env dir and subdirs will be affected in any way.

Reply:
I'm can do that.  👍 

I was erring on the side of explicitness, because this function looks like it has ability to write to the directory structure, and the thought of that worried *me*.  If it were a purely information-gathering API function I think I would've left that part of the sentence out.